src/tuning/tuning.cpp

```diff
@@ -289,56 +392,36 @@ void Tuner(int argc, char* argv[], const int V, GetTunerDefaultsFunc GetTunerDef
+                                  args.threads));
+  }
+
   // Starts the tuning process
   auto results = std::vector<TuningResult>();
   for (auto config_id = size_t{0}; config_id < configurations.size(); ++config_id) {
     try {
-      auto configuration = configurations[config_id];
-      printf("| %4zu | %5zu |", config_id + 1, configurations.size());
-      for (const auto& parameter : settings.parameters) {
-        printf("%5zu", configuration.at(parameter.first));
-      }
-      printf(" |");
-
       // Sets the input
       for (const auto id : settings.inputs) {
         device_buffers[id].Write(queue, buffer_sizes[id], source_buffers[id]);
       }
 
-      // Sets the thread configuration
-      auto global =
-          SetThreadConfiguration(configuration, settings.global_size, settings.mul_global, settings.div_global);
-      auto local = SetThreadConfiguration(configuration, settings.local_size, settings.mul_local, settings.div_local);
-
-      // Make sure that the global worksize is a multiple of the local
-      for (auto i = size_t{0}; i < global.size(); ++i) {
-        while ((global[i] / local[i]) * local[i] != global[i]) {
-          global[i]++;
-        }
-      }
-      if (local.size() > 1 && global.size() > 1) {
-        printf("%8zu%8zu |%8zu%8zu |", local[0], local[1], global[0], global[1]);
-      } else {
-        printf("%8zu%8d |%8zu%8d |", local[0], 1, global[0], 1);
+      Kernel kernel{nullptr};
+      std::vector<size_t> global;
+      std::vector<size_t> local;
+      {
+        std::unique_lock<std::mutex> lock(thread_infos[config_id].mtx);
+        thread_infos[config_id].cv.wait(lock, [&] { return thread_infos[config_id].ready; });
+        kernel = std::move(thread_infos[config_id].kernel);
+        global = std::move(thread_infos[config_id].global);
+        local = std::move(thread_infos[config_id].local);
+        printf("%s", thread_infos[config_id].print_info.c_str());
       }
 
-      // Sets the parameters for this configuration
-      auto kernel_source = std::string{""};
-      for (const auto& parameter : configuration) {
-        kernel_source += "#define " + parameter.first + " " + ToString(parameter.second) + "\n";
-      }
-      kernel_source += settings.sources;
-
-      // Compiles the kernel
-      const auto start_time = std::chrono::steady_clock::now();
-      auto compiler_options = std::vector<std::string>();
-      const auto program = CompileFromSource(kernel_source, args.precision, settings.kernel_name, device, context,
-                                             compiler_options, 0, true);
-      auto kernel = Kernel(program, settings.kernel_name);
-      const auto elapsed_time = std::chrono::steady_clock::now() - start_time;
-      const auto timing = std::chrono::duration<double, std::milli>(elapsed_time).count();
-      printf("   %sOK%s  %5.0lf ms |", kPrintSuccess.c_str(), kPrintEnd.c_str(), timing);
-
       // Runs the kernel
```

Comment:
Just making sure: running the kernel doesn't run in the parallel loop, right? Because on some devices launching multiple OpenCL kernels at the same time might influence performance numbers.